backend/src/v2/cmd/driver/main.go

```diff
@@ -292,5 +294,11 @@ func newMlmdClient() (*metadata.Client, error) {
 		mlmdConfig.Address = *mlmdServerAddress
 		mlmdConfig.Port = *mlmdServerPort
 	}
-	return metadata.NewClient(mlmdConfig.Address, mlmdConfig.Port)
+
+	mlPipelineServiceTLSEnabled, err := strconv.ParseBool(*mlPipelineServiceTLSEnabledStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return metadata.NewClient(mlmdConfig.Address, mlmdConfig.Port, mlPipelineServiceTLSEnabled)
```

Comment:
Related to my comment [here](https://github.com/opendatahub-io/data-science-pipelines/pull/72/files#r1747345833), I think we may have confused 2 things: 

KFP Server (http/grpc) is TLS Enabled == accepting tls connections
MLMD (grpc) Server is TLS Enabled == accepting tls connections 

These are not the same, and _**I think**_ here we are saying "If kfp server is tls enabled, let's configure our mlmd client assuming that is also tls enabled", which is not correct. We may need to introduce a new flag for specifically when mlmd is tls enabled.

Reply:
Well... Let's validate my understanding.

My understanding is that when [PodToPodTLS](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/20dd747d4342088a42dfce3cc98f907573ea3f82/api/v1alpha1/dspipeline_types.go#L51) is true, the communication between all pods should be through TLS. That includes the MLMD (grpc) server.
[This config](https://github.com/opendatahub-io/data-science-pipelines/pull/72#discussion_r1747345833) reflects [PodToPodTLS](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/20dd747d4342088a42dfce3cc98f907573ea3f82/api/v1alpha1/dspipeline_types.go#L51). (I should have used the same name)
That said, I didn't create a specific config for MLMD.

Is my understanding incorrect?

Reply:
Clarified offline.
I added a new flag for MLMD.